model/modelindexer/bulk_indexer.go

```diff
@@ -146,8 +146,9 @@ func (b *bulkIndexer) Flush(ctx context.Context) (elasticsearch.BulkIndexerRespo
 	}
 
 	req := esapi.BulkRequest{Body: &b.buf}
+	req.Header = esHeader
 	if b.gzipw != nil {
-		req.Header = gzipHeader
+		req.Header.Set("Content-Encoding", "gzip")
```

Comment:
Can we find a way to not modify the original `esHeader`? This approach will modify the underlying `esHeader` potentially sending the incorrect `Content-Encoding` when `b.gzipw` is nil. See https://go.dev/play/p/vh9diis70Uf.

Since these seem to be the only two options, how about creating 2 package vars, one with just `X-Elastic-Product-Origin`set  and a second one with `X-Elastic-Product-Origin` and `Content-Encoding` set? Perhaps not the nicest code but it saves us doing any copy and incurring unnecessary allocations. Also open to any suggestions on nicer approaches that you think are best.

Reply:
ahhh, i didn't even think of modifying the package-level map. great catch. yeah, the second map is cleaner and saves us the worry of mutation.